assets/js/modules/adsense/components/dashboard/AdBlockingRecoveryWidget.test.js

```diff
@@ -0,0 +1,299 @@
+	} );
+
+	afterEach( () => {
+		unsubscribeFromAll( registry );
+	} );
+
+	describe( 'before click', () => {
+		it.each( [
+			[
+				'the Adsense module is not connected',
+				ACCOUNT_STATUS_PENDING,
+				SITE_STATUS_READY,
+				'',
+				false,
+				false,
+				timestampThreeWeeksAgo,
+			],
+			[
+				'notification is dismissed',
+				ACCOUNT_STATUS_PENDING,
+				SITE_STATUS_READY,
+				'',
+				true,
+				true,
+				timestampThreeWeeksAgo,
+			],
+			[
+				'the Adsense account status is not ready',
+				ACCOUNT_STATUS_PENDING,
+				SITE_STATUS_READY,
+				'',
+				true,
+				false,
+				timestampThreeWeeksAgo,
+			],
+			[
+				'the Adsense site status is not ready',
+				ACCOUNT_STATUS_READY,
+				SITE_STATUS_ADDED,
+				'',
+				true,
+				false,
+				timestampThreeWeeksAgo,
+			],
+			[
+				'the Ad blocking recovery status is an empty string',
+				ACCOUNT_STATUS_READY,
+				SITE_STATUS_READY,
+				'',
+				true,
+				false,
+				timestampThreeWeeksAgo,
+			],
+			[
+				'the setup completed timestamp is less than three weeks',
+				ACCOUNT_STATUS_READY,
+				SITE_STATUS_READY,
+				AD_BLOCKING_RECOVERY_SETUP_STATUS_SETUP_CONFIRMED,
+				true,
+				false,
+				timestampLessThanThreeWeeksAgo,
+			],
+		] )(
+			'should not render the widget when %s',
+			(
+				testName,
+				accountStatus,
+				siteStatus,
+				adBlockingRecoverySetupStatus,
+				isModuleConnected,
+				isNotificationDismissed,
+				setupCompletedTimestamp
+			) => {
+				provideModules( registry, [
+					{
+						slug: 'adsense',
+						active: true,
+						connected: isModuleConnected,
+					},
+				] );
+
+				registry.dispatch( MODULES_ADSENSE ).receiveGetSettings( {
+					...validSettings,
+					accountStatus,
+					siteStatus,
+					adBlockingRecoverySetupStatus,
+					setupCompletedTimestamp,
+				} );
+				registry
+					.dispatch( CORE_USER )
+					.receiveGetDismissedItems( [
+						...( isNotificationDismissed
+							? [ 'ad-blocking-recovery-notification' ]
+							: [] ),
+					] );
+
+				const { container } = render(
+					<AdBlockingRecoveryWidget
+						Widget={ Widget }
+						WidgetNull={ WidgetNull }
+					/>,
+					{
+						registry,
+					}
+				);
+
+				expect( container ).toBeEmptyDOMElement();
+			}
+		);
+
+		it( 'should render the widget for the existing site without the setup completion time', () => {
+			registry
+				.dispatch( MODULES_ADSENSE )
+				.receiveGetSettings( validSettings );
+
+			const { container } = render(
+				<AdBlockingRecoveryWidget
+					Widget={ Widget }
+					WidgetNull={ WidgetNull }
+				/>,
+				{
+					registry,
+				}
+			);
+			expect( container ).toHaveTextContent(
+				'Recover revenue lost to ad blockers'
+			);
+		} );
+
+		it( 'should render the widget for the site with a setup completion time of more than three weeks', () => {
+			registry.dispatch( MODULES_ADSENSE ).receiveGetSettings( {
+				...validSettings,
+				setupCompletedTimestamp: 1684145866,
```

Comment:
Please can you also update this timestamp to use the appropriate semantic constant?